docs/blocks-storage/production-tips.md

```diff
@@ -58,3 +58,11 @@ The store-gateway stores each block’s index-header on the local disk and loads
 ### Ensure the compactor has enough disk space
 
 The compactor generally needs a lot of disk space in order to download source blocks from the bucket and store the compacted block before uploading it to the storage. Please refer to [Compactor disk utilization](./compactor.md#compactor-disk-utilization) for more information about how to do capacity planning.
+
+## Caching
+
+### Ensure memcached is properly scaled
+
+The rule of thumb to ensure memcached is probably scaled is to make sure evictions happen infrequently. When that's not the case and they affect query performances, the suggestion is to scale out the memcached cluster adding more nodes or increasing the memory limit of existing ones.
```

Comment:
The required cache is also dependent on the configured max lifetime of an object stored in memcached no? If so, what lifetime should we choose for each cache type?

Reply:
@weeco wrote:

> The required cache is also dependent on the configured max lifetime of an object stored in memcached no? If so, what lifetime should we choose for each cache type?

Various cache items come with default lifetimes already (see default values for chunks, metadata and index cache). While it’s possible to reconfigure it, changing it also requires knowledge about how individual items are used.

When block appears on the storage, it is only being used after meta.json file is created (this file is uploaded last). When deletion-marker file is added to the block, block is considered to be “soft-deleted” (such block can still be used for `ignore_deletion_mark_delay` duration). After `deletion_delay` passes (since timestamp stored in deletion marker), compactor will remove such soft-deleted blocks (deleting meta.json first).

Once block is fully uploaded (meta.json is uploaded last), files/objects inside block are immutable, except deletion marker may appear, and entire block may be removed (meta.json is removed first). “metafile” in metadata_cache config refers to these two files.

Hope this helps you in understanding and making cache config changes. [Worth including in the docs?]